### PR TITLE
Write LPNRegistryV1 contract tests

### PR DESCRIPTION
**Pull request by RyanRHall**
This PR introduces unit tests for the `LPNRegistryV1` contract. There a couple inconsequential source code changes that make testing easier but shouldn't have any affect on function. There are several notable TODOs still:

1. test the zk-validation path for `respond()`
2. edge cases and error case tests for `respond()`
3. Add more and bettor error cases (see `vm.skip(true)` tests)

This PR also adds a coverage ci action. The current amount of contract test coverage is north of 12%, and ideally we'd get this to 100% over time :).

### CODE REVIEW

File: test/v1/LPNRegistryV1.t.sol [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+        vm.expectRevert(QueryManager.QueryAfterCurrentBlock.selector);
+        registry.request{value: FEE}(
+            QUERY_HASH, PLACEHOLDERS, TEST_START_BLOCK, futureBlock
+        );
+    }
+
+    function test_respond_success() public {
+        // to properly test this we need to either:
+        //   a. mock the groth16 verifier library or ...
+        //   b. generate a real, working proof for this test (preferred)
```
Comment (nikkolasg) [inline test/v1/LPNRegistryV1.t.sol:221]: We can give you a real working proof but it wont necessarily follow the latest params onchain. If we could manage to have a test version of Groth16Verifier that is fixed (it only contains the cryptographic verification, no lagrange specific logic) then we could do it.

File: test/v1/LPNRegistryV1.t.sol [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+
+    function test_respond_invalidRequestId_reverts() public {
+        // New error required
+        vm.skip(true);
+        uint256 invalidRequestId = 9999; // Non-existent requestId
+        bytes32[] memory data;
+        uint256 blockNumber = block.number;
+        vm.prank(stranger);
+        // vm.expectRevert(QueryManager.UnknownRequestID.selector); // DNE
+        registry.respond(invalidRequestId, data, blockNumber);
```
Comment (nikkolasg) [inline test/v1/LPNRegistryV1.t.sol:233]: why shouldn't it revert ?

File: test/v1/LPNRegistryV1.t.sol (hunk 1 above)
Comment (RyanRHall) [reply test/v1/LPNRegistryV1.t.sol:221]: yeah I noticed there is a proof sitting in `test/v0/full_proof.bin`, but I imagine it is old and no longer functional. We need a way to test this without having to manually update a binary every time the groth verifier changes 🤔
